### PR TITLE
Use new method to get colormap

### PR DESCRIPTION
**Pull request by jl-wynen**
The old code will be deprecated in the future and caused a test failure in scippneutron

### CODE REVIEW

File: src/scipp/plotting/tools.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def parse_params(params=None, defaults=None, globs=None, array=None):
         parsed["cmap"] = LinearSegmentedColormap.from_list(
             "tmp", [parsed["color"], parsed["color"]])
     else:
-        parsed["cmap"] = copy(cm.get_cmap(parsed["cmap"]))
+        parsed["cmap"] = copy(matplotlib.colormaps[parsed["cmap"]])
```
Comment (nvaytet) [inline src/scipp/plotting/tools.py:88]: Note that this is not supported in mpl < 3.4 (I think). So either we do this and put a lower bound on the mpl requirement, or we filter out the warning for now.

Comment (jl-wynen) [reply src/scipp/plotting/tools.py:88]: We would have to filter the warning in downstream code as well, see scippneutron. Or we filter it just in this call until it becomes an actual deprecation warning. But then we have to revisit this. Is there a problem with a lower bound on mpl?

Comment (nvaytet) [reply src/scipp/plotting/tools.py:88]: I think lower bound is fine

File: setup.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def get_cmake_args():
           "test": ["pytest", "matplotlib", "xarray", "pandas", "pythreejs"],
           'all': ['h5py', 'scipy>=1.7.0', 'graphviz', 'pooch'],
           'interactive': [
-              'ipympl', 'ipython', 'ipywidgets', 'matplotlib', 'jupyterlab',
+              'ipympl', 'ipython', 'ipywidgets', 'matplotlib>=3.5', 'jupyterlab',
```
Comment (nvaytet) [inline setup.py:65]: I think you are missing a few places:
- `requirements/extra.in`
- `docs/environments/scipp.yml` (`matplotlib-base`)
- `docs/environments/developer.yml` (`matplotlib-base`)

Comment (jl-wynen) [reply setup.py:65]: Done, thanks!

**Review by nvaytet — APPROVED** [review]

**Comment by SimonHeybrock** [comment]
Have you checked if this causes compatibility issues, e.g., with Mantid or SasView?

**Comment by nvaytet** [comment]
That is a very good point. I just checked mantid, it pins to 3.5 in their conda build config, so should be fine. Don't know about sasview

**Comment by nvaytet** [comment]
Sasview pins to `matplotlib==3.4.3`, in their buildtools requirements (https://github.com/SasView/sasview/blob/main/build_tools/requirements.txt#L25), I don't know if it's pinning also the runtime dependency.

However, I now have a feeling that there are potentially many other packages that may pin something, and we should probably try to stay as flexible as possible? So I would say that we should remove the lower version bound and filter the warning instead?

**Comment by SimonHeybrock** [comment]
Instead of filtering the warning, you can `try:` or `hasattr` with the new way, and fall back to the old. Our CI won't get the warning, but users can still use older versions.

**Comment by jl-wynen** [comment]
Reverted the version pin and implemented the suggestion

File: src/scipp/plotting/tools.py [hunk 3]
```diff
@@ -45,12 +45,23 @@ def to_bin_edges(x, dim):
...
 
 
+def get_colormap(name):
+    """
+    Return a matplotlib colormap.
+    """
+    import matplotlib
+    if hasattr(matplotlib, 'colormaps'):
+        # This exists since matplotlib v3.5
+        return matplotlib.colormaps[name]
```
Comment (nvaytet) [inline src/scipp/plotting/tools.py:55]: Note that `matplotlib.colormaps[name]` doesn't work when `name` is `None`, while `get_cmap` does.
But we should be fine, as I think we are setting the `cmap` name from the config if it is `None`.

Comment (jl-wynen) [reply src/scipp/plotting/tools.py:55]: So should I change it or are you happy with the current state?

Comment (nvaytet) [reply src/scipp/plotting/tools.py:55]: tests passed, so let's say we are ok. If it breaks for users, they can report it and we will change it then.

**Review by nvaytet — APPROVED** [review]

**Pull request merged**